cmd/keeper/keeper.go

```diff
@@ -777,9 +777,14 @@ func (p *PostgresKeeper) postgresKeeperSM(pctx context.Context) {
 			if err = pgm.Start(); err != nil {
 				log.Errorf("failed to start postgres: %v", err)
 				return
-			} else {
-				started = true
 			}
+			if pgm.IsStreaming() != nil {
```

Comment:
nit: this test unfortunate is a bit misleading as it is not obvious just by reading the code whether this is error or another pointer so it's not obvious whether this means posgres is streaming or not:
```
if err := pgm.IsStreaming(); err != nil {
...
}
```
to make it more obvious even if the error is not used.

Also, can you explain the logic behind a bit since I'm not quite familiar with what's going on or needs to happen?
If the postgres (instance) is not streaming but was successfully stopped you consider what started?
And what does it achieve - e.g. by removing the `resyncIfNotReady` you revert the regression with the standbys inadvertently removing state?
I also don't see a point in setting `started = true` only to exit from the method as `started` is a local?

Reply:
The problem I am trying to solve here is when the replication slot is not created yet on master while standby goes trough this `postgresKeeperSM` function. `pgm.IsStreaming()` is checking streaming this PostgreSQL from master or not. Technically there could be more than just a replication slot issue with standby, but by the architectural design of stolon it is really hard to catch every possible problem. So, here I am trying to address only one.

`started` variable is initialized outside of if scope as far I can see, no?